internal/core/src/segcore/ChunkedSegmentSealedImpl.cpp

```diff
@@ -863,16 +863,17 @@ ChunkedSegmentSealedImpl::mask_with_delete(BitsetTypeView& bitset,
 
     auto bitmap_holder = std::shared_ptr<DeletedRecord::TmpBitmap>();
 
-    if (!is_sorted_by_pk_) {
-        bitmap_holder = get_deleted_bitmap(del_barrier,
-                                           ins_barrier,
-                                           deleted_record_,
-                                           insert_record_,
-                                           timestamp);
-    } else {
-        bitmap_holder = get_deleted_bitmap_s(
-            del_barrier, ins_barrier, deleted_record_, timestamp);
-    }
+    auto search_fn = [this](const PkType& pk, int64_t barrier) {
+        return this->search_pk(pk, barrier);
```

Comment:
what happend here? what if pk is not ordered?

should we do
        if (!is_sorted_by_pk_) {
            return insert_record_.search_pk(pk, timestamp);
        } else {
            return this.search_pk(pk, timestamp);
        }

Reply:
or if could be always better to put the is_sorted_by_pk_ fallback logic into this.search_pk

Reply:
I did this in search_pk.